### PR TITLE
fix(eraser): solves issues with eraser

### PR DESCRIPTION
**Pull request by NLueg**
Hi there! 

I struggled a lot with implementing an `undo` and `erase` feature for my canvas.
After research, I found issue #421 which already provides a well-working version.

This PR includes the suggested changes which are provided by @JoeLeung32 (btw. thank you so much!)

### CODE REVIEW

**Comment by UziTech** [comment]
This library is meant to be for signatures so an erase feature doesn't really make sense. If you want to create a drawing_pad library that does this that would make more sense but I don't think an erase feature will be added to this library.

**Comment by NLueg** [comment]
I know, that it's not the main focus of this plugin. But actually, you have a Stackblitz link for the ´Erase feature` at the top of the README.
Currently, this feature doesn't work although you advertise it there. We can close this PR if this is not intended but maybe you should make this clear in the README and remove the Stackblitz link.

After struggling a lot I just found the issue I mentioned and the fix from more than 4 years ago worked pretty well.

**Comment by UziTech** [comment]
@szimek what do you think about this feature? We would just need to add `compositeOperation` to `pointGroupOptions`, or we could abstract it with something like `isErase`?

File: src/signature_pad.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export interface Options extends Partial<PointGroupOptions> {
 
 export interface PointGroup extends PointGroupOptions {
   points: BasicPoint[];
+  color: string;
+  compositeOperation: GlobalCompositeOperation;
```
Comment (UziTech) [inline src/signature_pad.ts:50]: penColor is already part of PointGroupOptions. compositeOperation should be part of PointGroupOptions.

File: src/signature_pad.ts [hunk 2]
```diff
@@ -333,8 +335,14 @@ export default class SignaturePad extends SignatureEventTarget {
 
     const pointGroupOptions = this._getPointGroupOptions();
 
+    const ctx = (event.target as HTMLCanvasElement).getContext('2d');
     const newPointGroup: PointGroup = {
       ...pointGroupOptions,
+      color: this.penColor,
+      dotSize: this.dotSize,
+      minWidth: this.minWidth,
+      maxWidth: this.maxWidth,
```
Comment (UziTech) [inline src/signature_pad.ts:344]: These should already be in pointGroupOptions

File: src/signature_pad.ts [hunk 3]
```diff
@@ -582,6 +590,12 @@ export default class SignaturePad extends SignatureEventTarget {
             basicPoint.time,
           );
 
+          this.penColor = color;
+          this.minWidth = minWidth;
+          this.maxWidth = maxWidth;
```
Comment (UziTech) [inline src/signature_pad.ts:595]: Why are these set?

**Review by UziTech — CHANGES_REQUESTED** [review]

**Comment by NLueg** [comment]
@UziTech thanks for the fast review! Actually, I don't thought about the changes proposed in the issue. I refactored the code as you requested and verified it in my fork.
It still works perfectly as expected and is much less hacky. Thank you!

File: src/signature_pad.ts (hunk 3 above)
Comment (NLueg) [reply src/signature_pad.ts]: I just done copy + paste without thinking 😄

File: src/signature_pad.ts (hunk 1 above)
Comment (NLueg) [reply src/signature_pad.ts]: You're right! This is completely obsolet.
I just done copy + pasted without thinking 😄

**Review by UziTech — CHANGES_REQUESTED** [review]
Thanks for simplifying it! Can you write a new test with erase and fix current tests?

**Comment by NLueg** [comment]
> Thanks for simplifying it! Can you write a new test with erase and fix current tests?

I added a test that checks whether the set `compositeOperation` is applied to the canvas context. Do you think that is enough? Or what test do you want to have?

**Review by UziTech — APPROVED** [review]
LGTM

**Comment by yserlong** [comment]
Hello,
since this change my project won't build anymore. It throw an error telling it can't find the type : GlobalCompositeOperation.
Maybe it's a compatibility problem with the typescript version or something ?

**Comment by UziTech** [comment]
Looks like this may require Typescript 4.6 or higher

**Comment by root5427** [comment]
Hello. Looks like this PR broke the erase feature that was working fine by setting up the `globalCompositeOperation` of the canvas context. I applied the new `compositeOperation` of `SignaturePad` but it is not working as expected. Check this fiddle to make sure it doesn't work properly: https://jsfiddle.net/jq9cyzuc/.

I would appreciate any suggestions to get this all working, because I may not have figured out how to work with `compositeOperation` due to my poor knowledge of the Canvas API.

**Pull request merged**